layout/app/catalog/component.js

```diff
@@ -1,42 +1,90 @@
-import React, { PureComponent } from 'react';
-import PropTypes from 'prop-types';
+import React, { Component } from 'react';
 import debounce from 'lodash/debounce';
+import { toastr } from 'react-redux-toastr';
 
 // components
 import Layout from 'layout/layout/layout-app';
 import SearchInput from 'components/ui/SearchInput';
 import Spinner from 'components/ui/Spinner';
 import DatasetList from 'components/datasets/list';
+import Paginator from 'components/ui/Paginator';
+
+// services
+import { fetchDatasets } from 'services/dataset';
 
 // utils
 import { logEvent } from 'utils/analytics';
+import { DATASETS_PER_PAGE } from './constants';
+
+class CatalogLayout extends Component {
+  constructor(props) {
+    super(props);
 
-class CatalogLayout extends PureComponent {
-  static propTypes = {
-    list: PropTypes.array.isRequired,
-    loading: PropTypes.bool.isRequired,
-    getDatasets: PropTypes.func.isRequired,
-    setDatasetsSearch: PropTypes.func.isRequired
-  };
+    this.state = {
+      datasets: [],
+      loading: true,
+      search: null,
+      page: 1,
+      totalItems: 0
+    };
+
+    this.handlePageChange = this.handlePageChange.bind(this);
+  }
 
-  getSearchResults = debounce((value) => {
-    const { getDatasets } = this.props;
+  componentDidMount() {
+    this.getDatasets();
+  }
+
+  componentDidUpdate(nextProps, nextState) {
+    if (this.state.page !== nextState.page || this.state.search !== nextState.search) {
+      this.getDatasets();
+    }
+  }
 
+  getDatasets() {
+    const { page, search } = this.state;
+    this.setState({ loading: true });
+
+    fetchDatasets(
+      {
+        'page[size]': DATASETS_PER_PAGE,
+        'page[number]': page,
+        status: 'saved',
+        published: true,
+        application: process.env.APPLICATIONS,
+        language: 'en',
+        includes: ['metadata', 'widget', 'layer', 'vocabulary'].join(','),
+        search,
+        sort: 'name,description'
+      },
+      {},
+      true
+    )
+      .then((result) => {
+        const { datasets, meta } = result;
+        this.setState({
+          datasets,
+          totalItems: meta['total-items'],
+          loading: false
+        });
+      })
+      .catch((error) => {
+        toastr.error('There was an error with the request', error);
```

Comment:
At this point, `error` is probably an Error object (coming from `new Error`), so if you want to render something, call its `message` property instead. Otherwise, it will crash.